src/mbgl/util/mapbox.cpp

```diff
@@ -44,7 +44,13 @@ std::string normalizeSourceURL(const std::string& url, const std::string& access
         throw std::runtime_error("You must provide a Mapbox API access token for Mapbox tile sources");
     }
 
-    return baseURL + "v4/" + url.substr(protocol.length()) + ".json?access_token=" + accessToken + "&secure";
+    auto queryIdx = url.find("?");
+    std::string query = "";
+    if (queryIdx != std::string::npos) {
+        query = "&" + url.substr(queryIdx + 1, url.length() - queryIdx + 1);
```

Comment:
What will this code do if `?` is the last character in the URL? That  "should never happen" with `mapbox://` URLs but [as we've seen](https://github.com/mapbox/mapbox-gl-native/issues/5722), sometimes the supposedly impossible happens. Please add a test for this case and adjust the code if necessary.